### PR TITLE
Update the SLSA provenance predicate to v1

### LINKED ISSUE #200: Include SLSA provenance v1 predicate language bindings with pre-generated code?
Per [this discussion](https://github.com/in-toto/in-toto-golang/pull/221#discussion_r1172704744), it sounds like there's interest in including pre-generated Golang bindings for the SLSA provenance v1 predicate as part of our in-toto/attestation/ go package.

Since SLSA already defines a proto for the [provenance predicate](https://github.com/slsa-framework/slsa/blob/main/docs/provenance/schema/v1/provenance.proto), it would be preferable not to have to create/maintain a separate copy in this repo, but rather use this proto to pre-generate the Golang (and other) bindings.

Thoughts? @MarkLodato @TomHennen

### PR DESCRIPTION
**Pull request by marcelamelara**
Also adds the protobuf definition for the v1 predicate (per https://github.com/slsa-framework/slsa/issues/855).

Fixes #200

### CODE REVIEW

**Review by TomHennen — CHANGES_REQUESTED** [review]

File: protos/in_toto_attestation/predicates/provenance/v1/provenance.proto [hunk 1]
```diff
@@ -0,0 +1,42 @@
+syntax = "proto3";
```
[the remark this answers is not in the document]
Comment (marcelamelara) [reply protos/in_toto_attestation/predicates/provenance/v1/provenance.proto:1]: we certainly can

[the remark this answers is not in the document]
Comment (marcelamelara) [reply protos/in_toto_attestation/predicates/provenance/v1/provenance.proto:2]: done!

**Review by TomHennen — APPROVED** [review]
Thanks!

**Review by pxp928 — APPROVED** [review]
LGTM!

**Pull request merged**